File: frontend/src/services/knowledgeManagementService.js
import apiClient from './api';

export const getArticles = () => apiClient.get('/suite/knowledge-management/articles');
export const createArticle = (dto) => apiClient.post('/suite/knowledge-management/articles', dto);
export const getSops = () => apiClient.get('/suite/knowledge-management/sops');
export const createSop = (dto) => apiClient.post('/suite/knowledge-management/sops', dto);

File: backend/src/main/java/com/awais/hr/module/knowledge/service/KnowledgeManagementService.java
package com.awais.hr.module.knowledge.service;

import java.util.List;
import java.util.Map;

public interface KnowledgeManagementService {
    List<Map<String, Object>> getArticles();
    Map<String, Object> createArticle(Map<String, Object> body);
    List<Map<String, Object>> getSops();
    Map<String, Object> createSop(Map<String, Object> body);
}
